src/cls/IPM/ResourceProcessor/CPF.cls

```diff
@@ -0,0 +1,90 @@
+Include %IPM.Formatting
+
+Class %IPM.ResourceProcessor.CPF Extends %IPM.ResourceProcessor.Abstract
+{
+
+/// Comma-separated list of resource attribute names that this processor uses
+Parameter ATTRIBUTES As STRING = "Name,Directory,Phase,CustomPhase,When";
+
+/// Description of resource processor class (shown in UI)
+Parameter DESCRIPTION As STRING = "Merges the specified CPF file in the ""Initialize"" lifecycle phase.";
+
+/// Directory containing the CPF file to merge
+Property Directory As %IPM.DataType.ResourceDirectory [ InitialExpression = "cpf" ];
+
+/// FileN ame of the CPF merge file
+Property Name As %IPM.DataType.ResourceName [ Required ];
+
+/// The phase before/after which the CPF file should be merged
+Property Phase As %IPM.DataType.PhaseName [ InitialExpression = "Initialize" ];
+
+/// If provided, the Phase property will be ignored. The CPF merge will happen during this custom phase.
+Property CustomPhase As %IPM.DataType.CustomPhaseName;
+
+/// When to merge the CPF file: Before or After the specified phase. This only applies to the standard phases.
+Property When As %IPM.DataType.PhaseWhen [ InitialExpression = "Before" ];
+
+Method OnBeforePhase(pPhase As %String, ByRef pParams) As %Status
+{
+    If (..When = "Before") && (..Phase = pPhase) && (..CustomPhase = "") {
+        Quit ..DoMerge(.pParams)
+    }
+    Quit $$$OK
+}
+
+Method OnAfterPhase(pPhase As %String, ByRef pParams) As %Status
+{
+    If (..When = "After") && (..Phase = pPhase) && (..CustomPhase = "") {
+        Quit ..DoMerge(.pParams)
+    }
+    Quit $$$OK
+}
+
+Method DoMerge(ByRef pParams) As %Status
+{
+    Try {
+        Set verbose = $GET(pParams("Verbose"))
+        Set root = ..ResourceReference.Module.Root
+        Set sourcesRoot = ..ResourceReference.Module.SourcesRoot
+        // Use Construct first, rather than NormalizeFilename, so we don't have to deal with leading/trailing slashes
+        Set dir = $SELECT($$$isWINDOWS: $REPLACE(..Directory, "/", "\"), 1: ..Directory)
+        Set dir = ##class(%File).Construct(root, sourcesRoot, dir)
+        Set filename = ##class(%File).NormalizeFilename(..Name, dir)
+        If filename = "" {
+            $$$ThrowStatus($$$ERROR($$$GeneralError, $$$FormatText("CPF file '%1' not found in directory '%2'", ..Name, dir)))
+        } ElseIf '##class(%File).Exists(filename) {
+            $$$ThrowStatus($$$ERROR($$$GeneralError, $$$FormatText("CPF file '%1' not found", filename)))
```

Comment:
Probably worthwhile having the same error message to indicate which directory the code looked at for the cpf file